src/REPLMode/REPLMode.jl

```diff
@@ -374,7 +374,7 @@ end
 #############
 function do_cmd(repl::REPL.AbstractREPL, input::String; do_rethrow=false)
     if !isinteractive() && !TEST_MODE[] && !PRINTED_REPL_WARNING[]
-        @warn "The Pkg REPL interface is intended for interactive use, use with caution from scripts."
+        @warn "The Pkg REPL mode is intended for interactive use, use with caution from scripts, or use the functional API."
```

Comment:
This sentence is a bit awkward perhaps, maybe something like
```suggestion
        @warn "The Pkg REPL mode is intended for interactive use only and should be used with caution from scripts. It is recommended to use the functional API instead."
```

Reply:
I agree that's better. Thanks.

I kind of guessed at it before, but can you clarify exactly what is to be cautious about? That a script might break because the REPL mode might prompt a "user" for input? (e.g. what the flag is for in `apt-get -y`)

Reply:
The REPL interface might change in backward-incompatible ways.

Reply:
So can it just prescribe: "The Pkg REPL mode is intended for interactive use only, and should not be used from scripts. It is recommended to use the functional API instead."

Reply:
Sounds good to me.

Reply:
Done